core/templates/domain/skill/editable-skill-backend-api.service.ts

```diff
@@ -16,124 +16,142 @@
  * @fileoverview Service to send changes to a skill to the backend.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
-require('domain/skill/skill-domain.constants.ajs.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('EditableSkillBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService',
-  'EDITABLE_SKILL_DATA_URL_TEMPLATE', 'SKILL_DATA_URL_TEMPLATE',
-  function(
-      $http, $q, UrlInterpolationService,
-      EDITABLE_SKILL_DATA_URL_TEMPLATE, SKILL_DATA_URL_TEMPLATE) {
-    var _fetchSkill = function(skillId, successCallback, errorCallback) {
-      var skillDataUrl = UrlInterpolationService.interpolateUrl(
-        EDITABLE_SKILL_DATA_URL_TEMPLATE, {
-          skill_id: skillId
-        });
+import { SkillDomainConstants } from
+  'domain/skill/skill-domain.constants';
+import { EditableSkillResponseConfig, UpdateSkillPayload, ChangeList } from
+  'domain/skill/SkillDomain.types';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service.ts';
 
-      $http.get(skillDataUrl).then(function(response) {
-        var skill = angular.copy(response.data.skill);
-        var groupedSkillSummaryDicts = angular.copy(
-          response.data.grouped_skill_summaries);
-        if (successCallback) {
-          successCallback({
-            skill: skill,
-            groupedSkillSummaries: groupedSkillSummaryDicts
-          });
-        }
-      }, function(errorResponse) {
+@Injectable({
+  providedIn: 'root'
+})
+export class EditableSkillBackendApiService {
```

Comment:
FYI this service will be renamed to SkillBackendApiService as of PR #8353

Reply:
Should I rename this service in my pr too?

Reply:
Merged with develop now, these files have been renamed